client/rest/src/plugins/rosetta/nem/NemProxy.js

```diff
@@ -104,15 +104,7 @@ export default class NemProxy {
 		if (this.mosaicPropertiesMap.has(fullyQualifiedName))
 			return this.mosaicPropertiesMap.get(fullyQualifiedName);
 
-		const mosaicDefinitions = await this.fetchAll(
-			`namespace/mosaic/definition/page?namespace=${mosaicId.namespaceId}`,
-			100,
-			jsonObject => jsonObject.mosaic
-		);
-
-		const mosaicDefinition = mosaicDefinitions.find(definition => definition.id.name === mosaicId.name);
-		if (!mosaicDefinition)
-			throw RosettaErrorFactory.INTERNAL_SERVER_ERROR;
```

Comment:
Should we still not fail, or should the mosaic def always be there?

Reply:
in case of missing, NIS will return 404, which will be mapped by `global.fetch` to `result.ok === false`. 
thus, `NemProxy.fetch` will throw `RosettaErrorFactory.CONNECTION_ERROR`